### DISCUSSION TITLE
Error: ENOENT: no such file or directory

### DISCUSSION THREAD

**Discussion by officialrajdeepsingh** [avitorio/outstatic#117] [Q&A]
```
Error: ENOENT: no such file or directory, open '/home/rajdeepsingh/testing-stuff/outstatic-next/outstatic/content/metadata.json'
```

My environment variable file look like. In my project I cannot find `outstatic/content` folder and my dashboard work properly, I also create posts and pages collections. I al;so create content in posts and pages folder.

```
// .env.local

# https://github.com/settings/applications/new

OST_GITHUB_ID=c0be5****225e45

OST_GITHUB_SECRET=0bab23a44*****8d8edfee02

# random string min 32 chars


OST_TOKEN_SECRET=roFiycU2AIpCIsddBoR0D+

OST_REPO_SLUG=officialrajdeepsingh

# OPTIONAL
# If empty this will default to main
# OST_REPO_BRANCH=YOUR_GITHUB_REPOSITORY_BRANCH

# content out put path
OST_CONTENT_PATH=outstatic/content 

```
I use PNPM, Ubuntu and node (v18.13.0).

## List dependencies: 
```
    "@tiptap/pm": "^2.0.4",
    "@types/node": "20.4.5",
    "@types/react": "18.2.18",
    "@types/react-dom": "18.2.7",
    "autoprefixer": "10.4.14",
    "eslint": "8.46.0",
    "eslint-config-next": "13.4.12",
    "next": "13.4.12",
    "outstatic": "^0.0.38",
    "postcss": "8.4.27",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "tailwindcss": "3.3.3",
    "typescript": "5.1.
```

**Comment by avitorio** [comment]
Thanks for reaching out. Did you install outstatic in a clean nextjs install or through the deploy with vercel button?

**Reply by officialrajdeepsingh** [reply]
Yes, I can install the out static package with PNPM from scratch.

**Reply by avitorio** [reply]
Can you try clicking this button in your Dashboard?

![CleanShot 2023-08-04 at 13 17 20@2x](https://github.com/avitorio/outstatic/assets/1417109/5d654612-cb15-449b-b554-48c822f17293)

**Reply by avitorio** [reply]
@officialrajdeepsingh Please, let me know if the above works. Thanks!

**Reply by officialrajdeepsingh** [reply]
It is not work

**Reply by officialrajdeepsingh** [reply]
Maybe some problem with Environment variables, I test out static on local and cloud IDE with GitHub code spaces, but error can't resolve.

github repo https://github.com/officialrajdeepsingh/outstatic-cms-demo
```env
# https://github.com/settings/applications/new

# REQUIRED Environment variables
OST_GITHUB_ID=c04*****e45
OST_GITHUB_SECRET=0bab2******************&&&&&&&&&******d8edfee02
OST_TOKEN_SECRET=roFi***********CIsddBoR0D+
OST_REPO_SLUG=outstatic-cms-demo

# OPTIONAL Environment variables
OST_REPO_BRANCH=main 
OST_REPO_OWNER=officialrajdeepsingh 
# OST_CONTENT_PATH=outstatic/content # Where content is saved, if empty it defaults to outstatic/content
# OST_MONOREPO_PATH=apps/web # Specify monorepo folder of your Outstatic content
```

**Reply by officialrajdeepsingh** [reply]
outstaic dashboard work fine and in `app/page.tsx` fetch data does not work and show error. Checkout the video for better understand.

[out static error](https://github.com/avitorio/outstatic/assets/32416397/35281a36-d0f2-462b-b856-2d62932449c7)

**Reply by avitorio** [reply]
The error you are showing is because you don't have the `metadata.json` file inside of your outstatic/content folder. You need to click on "Rebuild metadata" and then git pull your repository, so that the file is loaded locally.

What happens when you click on "Rebuild Metadata"?

**Reply by at-Duke** [reply]
I have the same problem.
I added "outstatic" to an existing "nextjs14" project.
Admin Dashboard works with Github. All files are uploaded there.
How do I now access the information from Github.
No directories such as “outstatic/content” are created automatically locally.
What am I doing wrong?

**Reply by avitorio** [reply]
The dashboard only deals with files that are on Github. To have them locally you need to run `git pull`

**Reply by at-Duke** [reply]
Is there an integrated git pull as an action command when I save something in outstatic?

**Reply by avitorio** [reply]
Nope. Not currently, not even sure if that'd be possible to trigger a git pull from a web interface...
Sorry about that.
